advisor/src/main/kotlin/advisors/Osv.kt

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (C) 2022 EPAM Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.advisor.advisors
+
+import java.net.URI
+import java.time.Instant
+
+import org.ossreviewtoolkit.advisor.AbstractAdviceProviderFactory
+import org.ossreviewtoolkit.advisor.AdviceProvider
+import org.ossreviewtoolkit.clients.osv.Ecosystem
+import org.ossreviewtoolkit.clients.osv.OsvApiClient
+import org.ossreviewtoolkit.clients.osv.OsvService
+import org.ossreviewtoolkit.clients.osv.VulnerabilitiesForPackageRequest
+import org.ossreviewtoolkit.clients.osv.Vulnerability
+import org.ossreviewtoolkit.model.AdvisorCapability
+import org.ossreviewtoolkit.model.AdvisorDetails
+import org.ossreviewtoolkit.model.AdvisorResult
+import org.ossreviewtoolkit.model.AdvisorSummary
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.VulnerabilityReference
+import org.ossreviewtoolkit.model.config.AdvisorConfiguration
+import org.ossreviewtoolkit.utils.common.collectMessages
+import org.ossreviewtoolkit.utils.common.enumSetOf
+import org.ossreviewtoolkit.utils.common.withoutPrefix
+import org.ossreviewtoolkit.utils.ort.OkHttpClientHelper
+import org.ossreviewtoolkit.utils.ort.log
+
+/**
+ * An advice provider that obtains vulnerability from OSV.dev https://osv.dev/.
+ */
+class Osv(name: String, advisorConfiguration: AdvisorConfiguration) : AdviceProvider(name) {
+    class Factory : AbstractAdviceProviderFactory<Osv>("Osv") {
+        override fun create(config: AdvisorConfiguration): Osv =
+            Osv(providerName, config)
+    }
+
+    override val details: AdvisorDetails = AdvisorDetails(providerName, enumSetOf(AdvisorCapability.VULNERABILITIES))
```

Comment:
Nit: Drop the explicit type for this override.

Reply:
This would result in a hint from IntelliJ asking to make it explicit for API stability.

Reply:
Oh, I actually tried, and that hint didn't show up for me (which made sense to me as this is an overridden property). But maybe I was just too impatient. Or the Kotlin 1.7.10 IDE plugin that I just installed behaves differently in this regard.